TunnelKit/Sources/Protocols/OpenVPN/TLSBox.m

```diff
@@ -216,31 +220,47 @@ - (BOOL)startWithError:(NSError *__autoreleasing *)error
     if (self.securityLevel != TLSBoxDefaultSecurityLevel) {
         SSL_CTX_set_security_level(self.ctx, (int)self.securityLevel);
     }
-    if (!SSL_CTX_load_verify_locations(self.ctx, [self.caPath cStringUsingEncoding:NSASCIIStringEncoding], NULL)) {
-        ERR_print_errors_fp(stdout);
-        if (error) {
-            *error = TunnelKitErrorWithCode(TunnelKitErrorCodeTLSCertificateAuthority);
+
+    if (self.caPEM) {
```

Comment:
@roop I trusted you on this bit, but the replaced logic doesn't seem equivalent. The change broke TLS handshake on certain devices (Synology, pfSense etc.):

https://www.reddit.com/r/passepartout/comments/qvegv3/unable_to_connect_after_update_to_1170/

Reply:
@keeshux Sorry about the regression.

This is my understanding:
- Old: `SSL_CTX_load_verify_locations(ctx, caPath, NULL)` says that the file at caPath should be trusted when performing verification
- New: Use `SSL_CTX_get_cert_store` to get the trusted store and add the ca to it with `X509_STORE_add_cert`

The [man page](https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_get_cert_store.html) for `SSL_CTX_get_cert_store` says:
> Typically the trusted certificate store is handled indirectly via using SSL_CTX_load_verify_locations(3). Using the SSL_CTX_set_cert_store() and SSL_CTX_get_cert_store() functions it is possible to manipulate the X509_STORE object beyond the SSL_CTX_load_verify_locations(3) call.

That's why I thought this was the appropriate replacement.